nanoc-core/lib/nanoc/core/utils.rb

```diff
@@ -14,13 +14,25 @@ module Utils
       def self.expand_path_without_drive_identifier(file_name, dir_string)
         res = File.expand_path(file_name, dir_string)
 
-        if Nanoc::Core.on_windows?
+        if windows_fs?
           # On Windows, strip the drive identifier, e.g. `C:`.
           res = res.sub(/^[A-Z]:/, '')
         end
 
         res
       end
+
+      # Returns `true` if absolute file paths start with a drive identifier, like `C:`.
+      def self.windows_fs?
+        # NOTE: This isn’t memoized with ||= because @_windows_fs is a boolean.
+
+        return @_windows_fs if defined?(@_windows_fs)
```

Comment:
Ok, great, I think this works correctly. Just to get pedantic, because this is a module function it might be better to memoize with class variable `@@_windows_fs` so it only gets evaluated once instead of for every instance. And because the line before last line already evaluates to the return value it's possible to leave off the very last line. But feel free to ignore. :-)

Reply:
I’ve opted for a “class instance” variable, which will work as intended because the method already is a class/module method. Class variables in Ruby have odd behavior; see the [Ruby style guide entry on class variables](https://github.com/rubocop/ruby-style-guide?tab=readme-ov-file#no-class-vars).

I can indeed leave off the last line, but I like the explicitness of it. (And Rubocop does not complain about it, so I’m happy with it.)